File: pay/src/test/java/com/truenine/component/pay/service/impl/WeChatPayServiceTest.java
package com.truenine.component.pay.service.impl;

import cn.hutool.core.lang.Snowflake;
import cn.hutool.core.util.IdUtil;
import com.truenine.component.pay.Application;
import com.truenine.component.pay.api.model.request.CreateOrderApiRequestParam;
import com.truenine.component.pay.api.model.response.CreateOrderApiResponseResult;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
import org.testng.annotations.Test;

import java.math.BigDecimal;

@Slf4j
@SpringBootTest(classes = Application.class)
public class WeChatPayServiceTest extends AbstractTestNGSpringContextTests {

  @Autowired
  private WeChatPayService weChatPayService;

  @Test
  void testCreateOrder() {
    Snowflake snowflake = IdUtil.getSnowflake();
    CreateOrderApiRequestParam createOrderRequestParam = new CreateOrderApiRequestParam();
    createOrderRequestParam.setOrderId(snowflake.nextIdStr());
    createOrderRequestParam.setOpenId("oRYYL5H-IKKK0sHs1L0EOjZw1Ne4");
    createOrderRequestParam.setMoney(new BigDecimal("0.01"));
    createOrderRequestParam.setTitle("一斤菠萝");

    CreateOrderApiResponseResult order = weChatPayService.createOrder(createOrderRequestParam);
    log.debug(order.toString());
  }

}

File: pay/src/test/kotlin/com/truenine/component/pay/service/impl/WeChatPayServiceTest.kt
package com.truenine.component.pay.service.impl

import cn.hutool.core.util.IdUtil
import com.truenine.component.core.lang.LogKt
import com.truenine.component.pay.PayEntrance
import com.truenine.component.pay.models.request.CreateOrderRequestParam
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.annotation.Rollback
import org.springframework.test.context.testng.AbstractTestNGSpringContextTests
import org.testng.annotations.Test
import java.math.BigDecimal

@Rollback
@SpringBootTest(classes = [PayEntrance::class])
class WeChatPayServiceTest : AbstractTestNGSpringContextTests() {

  @Autowired
  private lateinit var weChatPayService: WeChatPayService;

  private val log = LogKt.getLog(this::class)

  @Test
  fun testCreateOrder() {
    val snowflake = IdUtil.getSnowflake()
    val createOrderRequestParam = CreateOrderRequestParam().apply {
      this.orderId = snowflake.nextIdStr()
      this.openId = snowflake.nextIdStr()
      this.title = "买一斤大菠萝"
      this.money = BigDecimal("0.01")
    }
    val createOrderResponseResult = weChatPayService.createOrder(createOrderRequestParam)
    log.debug(createOrderResponseResult.toString());
  }

}
